pages/api/auth.ts

```diff
@@ -1,16 +1,42 @@
 import { NextApiRequest, NextApiResponse } from 'next';
+import { createAppExtension } from '@lib/appExtensions';
 import { encodePayload, getBCAuth, setSession } from '../../lib/auth';
 
 export default async function auth(req: NextApiRequest, res: NextApiResponse) {
-    try {
-        // Authenticate the app on install
-        const session = await getBCAuth(req.query);
-        const encodedContext = encodePayload(session); // Signed JWT to validate/ prevent tampering
+  try {
+  // Authenticate the app on install
+  const session = await getBCAuth(req.query);
+  const encodedContext = encodePayload(session); // Signed JWT to validate/ prevent tampering
+  const { access_token: accessToken, context } = session;
 
-        await setSession(session);
-        res.redirect(302, `/?context=${encodedContext}`);
-    } catch (error) {
-        const { message, response } = error;
-        res.status(response?.status || 500).json({ message });
-    }
+  const storeHash = context?.split('/')[1] || '';
+
+  await setSession(session);
+
+  /**
+   * For stores that do not have the app installed yet, create App Extensions when app is
+   * installed.
+   */
+
+  const requestBody = await createAppExtension();
+
+  const response = await(fetch(`https://api.bigcommerce.com/stores/${storeHash}/graphql`, {
+    method: "POST",
+        headers: {
+        accept: "application/json",
+        "content-type": "application/json",
+        "x-auth-token": accessToken,
+        },
+        body: JSON.stringify(requestBody),
+  }));
+  const { data, errors } = await response.json();
+  if(data === null && errors[0]?.message){
```

Comment:
qq: is `errors` an empty array if there are no errors? wondering if we could just check the length of errors here

Reply:
Oh yeah it is an empty array- checking the length makes sense with that!